### PR TITLE
remove torch from package exporter

### PR DESCRIPTION
**Pull request by PaliC**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #72246
* __->__ #72242
* #72563
* #72237

Removes the dependency of torch on torch.package.package_exporter.

Differential Revision: [D33973987](https://our.internmc.facebook.com/intern/diff/D33973987)

### CODE REVIEW

**Comment by PaliC** [comment]
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33973987).

File: torch/package/package_exporter.py [hunk 1]
```diff
@@ -23,7 +23,7 @@
 )
 
 import torch
```
Comment (suo) [inline torch/package/package_exporter.py:25]: shouldn't we actually be removing torch lol

File: torch/package/package_exporter.py [hunk 2]
```diff
@@ -850,14 +850,6 @@ def deny(self, include: "GlobPattern", *, exclude: "GlobPattern" = ()):
 
     def _persistent_id(self, obj):
         if hasattr(obj, "__reduce_package__") and not inspect.isclass(obj):
-            if _gate_torchscript_serialization and isinstance(
```
Comment (suo) [inline torch/package/package_exporter.py:853]: if we are removing this check, shouldn't we remove the `_gate_torchscript_serialization` variable as well?

**Review by suo — CHANGES_REQUESTED** [review]
some small cleanups but overall lgtm

File: torch/package/package_exporter.py (hunk 1 above)
Comment (PaliC) [reply torch/package/package_exporter.py:25]: Indeed haha

File: torch/package/package_exporter.py (hunk 2 above)
Comment (PaliC) [reply torch/package/package_exporter.py:853]: Yeah, I wanted to ask what was this for? Do we need to keep this in an internal version?

**Comment by PaliC** [comment]
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33973987).

File: torch/package/package_exporter.py [hunk 3]
```diff
@@ -690,7 +687,7 @@ def register_extern_hook(self, hook: ActionHook) -> RemovableHandle:
         Hooks will be called in order of registration.
 
         Returns:
-            :class:`torch.utils.hooks.RemovableHandle`:
+            :class:`torch.package._hooks.RemovableHandle`:
```
Comment (suo) [inline torch/package/package_exporter.py:690]: This class should be a part of the public API, and thus should not have any underscores in its qualified name. The easiest way to do this is to re-export the name in `torch/pacakge/__init__.py` and reference it as `torch.package.RemovableHandle`.

**Review by suo — APPROVED** [review]
Small nit, otherwise lgtm

**Comment by PaliC** [comment]
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33973987).

**Pull request closed without merging**